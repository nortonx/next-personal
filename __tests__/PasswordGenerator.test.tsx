import { render, screen, fireEvent } from '@testing-library/react'
import PasswordGeneratorPage from '@/app/password-generator/page'

describe("Password Generator Page", () => {
  it("Should render page and match snapshot", async () => {
    render(<PasswordGeneratorPage/>)
    const page = await screen.findByTestId('password-generator')
    expect(page).toBeInTheDocument()
    expect(page).toMatchSnapshot()
  })
})