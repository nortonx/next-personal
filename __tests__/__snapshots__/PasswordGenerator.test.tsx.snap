// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Generator Page Should render page and match snapshot 1`] = `
<main
  data-testid="password-generator"
>
  <div
    class="container-fluid"
  >
    <div
      class="justify-content-md-center mb-4 row"
    >
      <div
        class="col-md-auto"
      >
        <p
          class="fs-2"
        >
          Password Generator
        </p>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
          >
            Generated password
          </label>
          <input
            class="form-control"
            type="text"
            value=""
          />
        </div>
        <div
          class="d-flex justify-content-around"
        >
          <button
            class="btn btn-warning"
            data-testid="reset-password"
            type="button"
          >
            Reset
          </button>
          <button
            class="btn btn-secondary"
            data-testid="copy-password"
            type="button"
          >
            Copy value
          </button>
        </div>
      </div>
    </div>
    <div
      class="justify-content-md-center mb-4 row"
    >
      <div
        class="col-md-auto"
      >
        <div
          class="my-2 form-check form-switch"
        >
          <input
            class="form-check-input"
            type="checkbox"
          />
          <label
            class="form-check-label"
            title=""
          >
            Enable numbers
          </label>
        </div>
        <div
          class="my-2 form-check form-switch"
        >
          <input
            class="form-check-input"
            type="checkbox"
          />
          <label
            class="form-check-label"
            title=""
          >
            Enable Uppercase Letters
          </label>
        </div>
        <div
          class="my-2 form-check form-switch"
        >
          <input
            class="form-check-input"
            type="checkbox"
          />
          <label
            class="form-check-label"
            title=""
          >
            Enable Special Characters
          </label>
        </div>
      </div>
    </div>
    <div
      class="justify-content-md-center mb-4 row"
    >
      <div
        class="col-md-auto"
      >
        <label
          class="form-label"
        >
          Password length
          : 
          8
        </label>
        <input
          class="form-range"
          max="18"
          min="1"
          step="1"
          type="range"
          value="8"
        />
      </div>
    </div>
    <div
      class="justify-content-md-center mb-4 row"
    >
      <div
        class="d-flex justify-content-between col-md-auto"
      >
        <button
          class="btn btn-outline-primary"
          type="button"
        >
          Generate Password
        </button>
      </div>
    </div>
    <div
      class="justify-content-md-center row"
    >
      <div
        class="col-md-auto"
      />
    </div>
  </div>
</main>
`;
